src/addon/info.cpp

```diff
@@ -132,6 +151,68 @@ std::string addon_info::display_title() const
 	}
 }
 
+addon_info_translation addon_info_translation::invalid = {"en_US", false, "invalid_addon!", ""};
+
+addon_info_translation addon_info::translated_info() const
+{
+	std::string locale = get_language().localename;
+
+	if(locale != "en_US") {
+		for(const addon_info_translation& info : this->info_translations) {
+			if(info.language == locale || info.language == locale.substr(0, 2)) {
+				return info;
+			}
+		}
+	}
+
+	return addon_info_translation::invalid;
+}
+
+std::string addon_info::display_title_translated() const
+{
+	addon_info_translation info = this->translated_info();
+
+	if(info.valid()) {
+		return info.title;
+	}
+
+	return "";
+}
+
+std::string addon_info::display_title_translated_or_original() const
+{
+	std::string title = display_title_translated();
+	return title.empty() ? display_title() : title;
+}
+
+std::string addon_info::description_translated() const
+{
+	addon_info_translation info = this->translated_info();
+
+	if(info.valid()) {
+		return info.description;
+	}
+
+	return this->description;
+}
+
+std::string addon_info::display_title_full() const
+{
+	std::string local_title = display_title_translated();
+	if(local_title.empty())
+		return display_title();
+	return local_title + " (" + display_title() + ")";
+}
+
+std::string addon_info::display_title_full_shift() const
+{
+	std::string local_title = display_title_translated();
+	if(local_title.empty())
+		return display_title();
+	return local_title + "\n"
+		+ "<small>(" + display_title() + ")</small>";
+}
```

Comment:
I'm not entirely sure these methods belong here, at least not in the current form. Both revolve around GUI implementation details, especially `display_title_full_shift()`, whose purpose is not self-evident from the name and there are no Doxygen comments attached to it. The second method at least seems like it belongs somewhere deep in the GUI2 Add-on Manager dialog's implementation instead.

(I'm also not entirely sure if using a linebreak and Pango markup to achieve whatever the method is trying to achieve here *is* a good idea in the first place, that's more @Vultraz's thing I guess.)

Reply:
Well, `display_title_translated_or_original()` and `display_title_full_shift()` can be moved to the classes where they are referrenced. However, `display_title_full()` is used in quite a few various places to indicate the name of the addon, so I doubt if I should move it somewhere.

Reply:
I agree `display_title_full()` probably should remain where it is since it's used by the add-ons client class.